src/Apps/Settings/Routes/Shipping/Components/SettingsShippingAddressForm.tsx

```diff
@@ -314,22 +297,39 @@ export const SettingsShippingAddressForm: FC<SettingsShippingAddressFormProps> =
 
                 <Column span={12}>
                   <PhoneNumberInput
-                    phoneNumber={{
-                      isValid: isEditing,
-                      national: values.attributes.phoneNumber,
-                      regionCode: values.attributes.phoneNumberCountryCode,
+                    onValidation={result => {
+                      setFieldValue(
+                        "attributes.isPhoneNumberValid",
+                        result.isValid
+                      )
+
+                      if (result.isValid) {
+                        setFieldValue("attributes.phoneNumber", result.national)
+                      }
                     }}
-                    onPhoneNumberValidation={handlePhoneNumberValidation}
                     inputProps={{
-                      maxLength: 25,
-                      onBlur: handleBlur("attributes.phoneNumber"),
+                      name: "attributes.phoneNumber",
                       placeholder: "(000) 000 0000",
+                      onChange: handleChange,
+                      onBlur: handleBlur,
+                      value: values.attributes.phoneNumber,
```

Comment:
Either way it's much easier to avoid callback loops by simply using Formik's state management